covalent_ssh_plugin/ssh.py

```diff
@@ -116,128 +115,29 @@ def __init__(
-                    output, stdout, stderr, exception = self._on_ssh_fail(fn, args, kwargs, stdout, stderr, message)
-                    if exception:
-                        raise exception
-                    return (output, stdout, stderr)
-
-            cmd = f"mkdir -p {self.remote_cache_dir}"
-            client_in, client_out, client_err = self.client.exec_command(cmd)
-            err = client_err.read().decode("utf8")
-            if err != "":
-                app_log.warning(err)
-
-            # Pickle and save location of the function and its arguments:
-            self._write_function_files(operation_id, fn, args, kwargs)
-
-            # scp pickled function and run script to server here:
-            scp = SCPClient(self.client.get_transport())
-            scp.put(self.function_file, self.remote_function_file)
-            scp.put(self.script_file, self.remote_script_file)
-
-            # Run the function:
-            cmd = f"{self.python3_path} {self.remote_script_file}"
-            client_in, client_out, client_err = self.client.exec_command(cmd)
-            err = client_err.read().decode("utf8").strip()
-            if err != "":
-                app_log.warning(err)
-
-            # Check that a result file was produced:
-            cmd = f"ls {self.remote_result_file}"
-            client_in, client_out, client_err = self.client.exec_command(cmd)
-            if client_out.read().decode("utf8").strip() != self.remote_result_file:
-                message = f"Result file {self.remote_result_file} on remote host {self.hostname} was not found"
-                output, stdout, stderr, exception = self._on_ssh_fail(fn, args, kwargs, stdout, stderr, message)
-                if exception:
-                    raise exception
-                return (output, stdout, stderr)
-
-            # scp the pickled result to the local machine here:
-            result_file = os.path.join(self.cache_dir, f"result_{operation_id}.pkl")
-            scp.get(self.remote_result_file, result_file)
-
-            # Load the result file:
-            with open(result_file, "rb") as f_in:
-                result, exception = pickle.load(f_in)
-
-            if exception is not None:
-                app_log.debug(f"exception: {exception}")
-                raise exception
-
-        self.client.close()
-
-        if info_queue:
-            # Update the status:
-            info_dict = info_queue.get()
-            info_dict["STATUS"] = Result.FAILED if result is None else Result.COMPLETED
-            info_queue.put(info_dict)
-
-        # TODO: fix execution._run_task() to parse exception as in covalent-microservices
-        # return (result, stdout.getvalue(), stderr.getvalue(), exception)
-        return (result, stdout.getvalue(), stderr.getvalue())
+        self.dispatch_id = dispatch_id
+        self.node_id = node_id
+        return await super().execute(function, args, kwargs, dispatch_id, results_dir, node_id)
```

Comment:
is this call needed? I understood that the `BaseAsyncExecutor` would invoke the setup, run and teardown methods

Reply:
In view of the revamped BaseAsyncExecutor API, we should just use `run()` here

Reply:
> is this call needed? I understood that the `BaseAsyncExecutor` would invoke the setup, run and teardown methods

Yes, it would, but if you see the code in `run`, you'll see it needs the `dispatch_id` and `node_id`, which is only provided to the executor here, which means that this is the only natural place for it to be set.

> In view of the revamped BaseAsyncExecutor API, we should just use `run()` here

We are still using `run()` for the actual "running" code. In this particular case, `run()` requires `dispatch_id` and `node_id`, which can easily be passed to it as class variables. If I understood it correctly then the purpose of `run` is to make things easier for new executor creators. This executor is slightly more advanced which is why there is a need to override `execute` as well.

Reply:
> In this particular case, run() requires dispatch_id and node_id, which can easily be passed to it as class variables. 

That's a good point. `dispatch_id` and `node_id` will be needed for a typical remote executor. @santoshkumarradha should we open a quick PR to store the these attributes in the executor instance?

Reply:
@cjao How do you propose we do that other than this way. I found this to be the most natural way because this is the place where we are already getting the `dispatch_id` and `node_id` and we are simply assigning it here.

Reply:
Good discussion, Will get back shortly after a quick chat with the team !

Reply:
@kessler-frost I have no objection to what you're doing right now.

Reply:
Ah, got it!

> improvements in the BaseAsyncExecutor interface.

Do you mean the setup and teardown ones?

Reply:
Yes that makes sense, Its ✅ to be added as a backlog for base executer. We can come back to changing the base/plugins with the attribute.